build: disable v8 snapshot compression

joyeecheung opened the pull request:
Following https://chromium-review.googlesource.com/c/v8/v8/+/3275554. Locally the binary size is increased by ~2.4mb (+2.9%) with a pretty significant speedup in startup:

```
                                                                                     confidence improvement accuracy (*)   (**)  (***)
misc/startup.js count=30 mode='process' script='benchmark/fixtures/require-builtins'        ***     12.28 %       ±1.13% ±1.52% ±2.00%
misc/startup.js count=30 mode='process' script='test/fixtures/semicolon'                    ***     17.39 %       ±1.12% ±1.50% ±1.97%
misc/startup.js count=30 mode='worker' script='benchmark/fixtures/require-builtins'         ***     14.54 %       ±0.77% ±1.03% ±1.35%
misc/startup.js count=30 mode='worker' script='test/fixtures/semicolon'                     ***     18.02 %       ±1.02% ±1.36% ±1.79%
```

I think the increased binary size is worth it, but if we really worry about it, another (perhaps better) option would be to disable the built-in compression of V8 and compress the snapshot + source code + code cache ourselves using a different compression library (e.g. Deno switched to [lz4 and zstd](https://github.com/denoland/deno/pull/13320) and got a pretty significant performance boost). I'd suggest to just disable it for now, regardless of whether we can follow up with a compression strategy that can reduce the binary size significantly without hurting performance, because currently a 2.9% binary size saving is not really worth the performance regression.

<!--
Before submitting a pull request, please read
https://github.com/nodejs/node/blob/HEAD/CONTRIBUTING.md.

Commit message formatting guidelines:
https://github.com/nodejs/node/blob/HEAD/doc/contributing/pull-requests.md#commit-message-guidelines

For code changes:
1. Include tests for any bug fixes or new features.
2. Update documentation if relevant.
3. Ensure that `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
-->

joyeecheung commented:
CI: https://ci.nodejs.org/view/Node.js%20benchmark/job/benchmark-node-micro-benchmarks/1253/

mscdex commented:
Maybe we could add a build option to enable it for those who want to optimize for size (along with other build options)?

joyeecheung commented:
Rebased against https://github.com/nodejs/node/pull/45746 to pull in better benchmarks and added the build option to enable the compression thanks to suggestion from @mscdex 

Locally the size increase is now 2.4MB from the main branch, with the following benchmark numbers:

```
                                                                                     confidence improvement accuracy (*)   (**)  (***)
misc/startup.js count=30 mode='process' script='benchmark/fixtures/require-builtins'        ***     12.28 %       ±1.13% ±1.52% ±2.00%
misc/startup.js count=30 mode='process' script='test/fixtures/semicolon'                    ***     17.39 %       ±1.12% ±1.50% ±1.97%
misc/startup.js count=30 mode='worker' script='benchmark/fixtures/require-builtins'         ***     14.54 %       ±0.77% ±1.03% ±1.35%
misc/startup.js count=30 mode='worker' script='test/fixtures/semicolon'                     ***     18.02 %       ±1.02% ±1.36% ±1.79%
```

New CI: https://ci.nodejs.org/view/Node.js%20benchmark/job/benchmark-node-micro-benchmarks/1254/

joyeecheung commented:
Landed in a996638e53c82a7c60589f88c6d7b517e9fd5505

kkocdko commented:
Is there any plan to enable snapshot compression with lz4 or zstd?
